### PR TITLE
gh-125741: Update check_generated_files CI to use our published container image

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-125741

### CODE REVIEW

**Comment by corona10** [comment]
@erlend-aasland  @hugovk 

I verified the the workflows well through https://github.com/python/cpython/actions/runs/11422707050/job/31781018424
This PR separates the task from `check_generated_files` but for unifying to the single task, we may need full unified devcontainer images to be used at https://github.com/python/cpython-devcontainers

Let's handle it separately.

**Comment by Damien-Chen** [comment]
Since the container python used has something different from Github Action official base image and that is the reason why we should switch to the internal container ?

**Comment by corona10** [comment]
See: https://discuss.python.org/t/pinning-gnu-autotools-using-the-github-container-registry/47437/29

FYI, The purpose of CI is verification of whether the person generate autoconf properly

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks. I need this for https://github.com/python/cpython/pull/125571.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -40,6 +40,50 @@ jobs:
...
+    # Don't use ubuntu-latest but a specific version to make the job
+    # reproducible: to get the same tools versions (autoconf, aclocal, ...)
+    runs-on: ubuntu-24.04
+    container:
+      image: ghcr.io/python/autoconf:2024.10.11.11293396815
+    timeout-minutes: 60
+    needs: check_source
+    if: needs.check_source.outputs.run_tests == 'true'
+    steps:
+      - name: Install git
```
Comment (hugovk) [inline .github/workflows/build.yml]: Is Git not already part of the image? Should it be? 

If it needs adding, can be done in a followup. 

---

Git nit:


```suggestion
      - name: Install Git
```

**Review by hugovk — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (corona10) [reply .github/workflows/build.yml]: In the future image I will add it. Calver makes the some trouble about old OS image :) But based on new image or our unified image will solve this issue

**Comment by Damien-Chen** [comment]
Does that should be backport to 3.13 and 3.12 ?

**Comment by erlend-aasland** [comment]
> Does that should be backport to 3.13 and 3.12 ?

I think we should backport it, yes. Hugo?

**Comment by hugovk** [comment]
Yes, it'll be easier to maintain if the three branches are more similar.

**Pull request merged**